In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
delays = pd.read_excel("Простои за июль 2024.xlsx")
sensor_data = pd.read_excel("Датчики за июль 2024.xlsx")

In [3]:
delays.head()

start                 end   duration  \
0 2024-07-01 11:35:08 2024-07-01 11:55:33  20.416666   
1 2024-07-01 15:52:59 2024-07-01 16:14:41  21.700000   
2 2024-07-01 17:52:22 2024-07-01 18:21:48  29.433333   
3 2024-07-01 18:22:22 2024-07-01 18:31:06   8.733333   
4 2024-07-01 18:51:42 2024-07-01 18:54:03   2.350000   

                     downtime                        reason  \
0                 Переналадки  Переход продукт вода продукт   
1                 Переналадки  Переход продукт вода продукт   
2  Прочие внеплановые простои          Загруженность склада   
3  Прочие внеплановые простои          Загруженность склада   
4  Прочие внеплановые простои          Загруженность склада   

                          description  reason_id detailed_description  \
0  Переход продукт вода продукт + АПМ    2214102              переход   
1  Переход продукт вода продукт + АПМ    2214102              переход   
2                             Съем ГП    2112001            нет сьема   
3                             Съем ГП    2112001            нет сьема   
4                             Съем ГП    2112001            нет сьема   

  component  
0    Розлив  
1    Розлив  
2    Розлив  
3    Розлив  
4    Розлив

In [4]:
sensor_data.head()

start                 end   duration  \
0 2024-07-01 08:00:00 2024-07-01 08:13:00  13.400000   
1 2024-07-01 08:13:00 2024-07-01 08:13:00   0.183333   
2 2024-07-01 08:13:00 2024-07-01 08:18:00   4.433333   
3 2024-07-01 08:18:00 2024-07-01 09:17:00  59.100000   
4 2024-07-01 09:17:00 2024-07-01 09:18:00   1.016666   

                      reason  
0         Производство > 95%  
1  Вмешательство оператора\n  
2   Производство > 75% < 95%  
3         Производство > 95%  
4             Внешняя ошибка

In [10]:
delays.info(), sensor_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 346 entries, 0 to 345
Data columns (total 9 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   start                 346 non-null    datetime64[ns]
 1   end                   346 non-null    datetime64[ns]
 2   duration              346 non-null    float64       
 3   downtime              346 non-null    object        
 4   reason                346 non-null    object        
 5   description           346 non-null    object        
 6   reason_id             346 non-null    int64         
 7   detailed_description  346 non-null    object        
 8   component             346 non-null    object        
dtypes: datetime64[ns](2), float64(1), int64(1), object(5)
memory usage: 24.5+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6644 entries, 0 to 6643
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype         
---  ------    ---------

(None, None)

In [15]:
sensor_data["start"] = sensor_data["start"].apply(lambda x: x.replace(second=0))
sensor_data["end"] = sensor_data["end"].apply(lambda x: x.replace(second=0))

delays["start"] = delays["start"].apply(lambda x: x.replace(second=0))
delays["end"] = delays["end"].apply(lambda x: x.replace(second=0))

In [ ]:
delay_full_data = pd.merge(sensor_data, delays, how="inner", on=["start", "end", "duration"])
delay_full_data

start                 end    duration  \
0   2024-07-01 11:35:00 2024-07-01 11:55:00   20.416666   
1   2024-07-01 15:52:00 2024-07-01 16:14:00   21.700000   
2   2024-07-01 17:52:00 2024-07-01 18:21:00   29.433333   
3   2024-07-01 18:22:00 2024-07-01 18:31:00    8.733333   
4   2024-07-01 18:51:00 2024-07-01 18:54:00    2.350000   
..                  ...                 ...         ...   
341 2024-07-30 19:11:00 2024-07-30 19:15:00    4.383333   
342 2024-07-30 21:05:00 2024-07-30 21:09:00    3.916666   
343 2024-07-30 21:42:00 2024-07-31 00:08:00  146.950000   
344 2024-07-31 00:09:00 2024-07-31 00:11:00    2.350000   
345 2024-07-31 01:22:00 2024-07-31 08:00:00  397.883333   

                      reason_x                    downtime  \
0    Вмешательство оператора\n                 Переналадки   
1    Вмешательство оператора\n                 Переналадки   
2              Затор на выходе  Прочие внеплановые простои   
3              Затор на выходе  Прочие внеплановые простои   
4    Вмешательство оператора\n  Прочие внеплановые простои   
..                         ...                         ...   
341             Внешняя ошибка       Санитарные регламенты   
342             Внешняя ошибка             Организационные   
343          Нехватка на входе                 Переналадки   
344  Вмешательство оператора\n                 Переналадки   
345                    Останов           Недоступное время   

                         reason_y                         description  \
0    Переход продукт вода продукт  Переход продукт вода продукт + АПМ   
1    Переход продукт вода продукт  Переход продукт вода продукт + АПМ   
2            Загруженность склада                             Съем ГП   
3            Загруженность склада                             Съем ГП   
4            Загруженность склада                             Съем ГП   
..                            ...                                 ...   
341                 Внешние мойки     Ополаскивание укупора и розлива   
342          Проверки по качеству               Лабораторные проверки   
343          Переход через щ. СИП              Продукт СИП щ. продукт   
344          Переход через щ. СИП              Продукт СИП щ. продукт   
345             Отсутствие заявок                   Отсутствие заявок   

     reason_id         detailed_description            component  
0      2214102                      переход               Розлив  
1      2214102                      переход               Розлив  
2      2112001                    нет сьема               Розлив  
3      2112001                    нет сьема               Розлив  
4      2112001                    нет сьема               Розлив  
..         ...                          ...                  ...  
341    2310502              Обл.гор. водой   Автомат укупорочный  
342    2512701                  Отбор проб.               Розлив  
343    2214201  Переход на FLash. СИП. АПМ.               Розлив  
344    2214201  Переход на FLash. СИП. АПМ.               Розлив  
345    2913801           Отсутствие заявок.               Розлив  

[346 rows x 10 columns]

In [ ]:
delay_full_data["reason_x"].value_counts()

reason_x
Вмешательство оператора\n    189
Внешняя ошибка                82
Нехватка на входе             37
Затор на выходе               14
Останов                       12
Ручное управление              6
Аварийная остановка            5
Внутренняя ошибка\n            1
Name: count, dtype: int64

In [36]:
delay_full_data["reason_y"].value_counts()

reason_y
Переход продукт вода продукт          81
Внешние мойки                         79
Переход через щ. СИП                  43
Проверки по качеству                  23
Перерывы                              21
Отсутствие заявок                     20
Переход продукт на продукт            17
Комментируемые остановки              13
Смена бутылки/банки                   13
Переход через стерил.                  8
Смена этикетки                         8
ПТО                                    6
Загруженность склада                   5
Тесты материалов                       3
Недостаточное количество персонала     2
Втор. упак.                            2
Переход через щ-к СИП                  1
Поддержка производства                 1
Name: count, dtype: int64

In [37]:
delay_full_data[delay_full_data["reason_x"] == "Аварийная остановка"]["reason_y"].value_counts()

reason_y
Внешние мойки               2
Комментируемые остановки    2
Переход через щ. СИП        1
Name: count, dtype: int64

In [38]:
delay_outer_data = pd.merge(sensor_data, delays, how="outer", on=["start", "end", "duration"])
delay_outer_data

start                 end    duration  \
0    2024-07-01 08:00:00 2024-07-01 08:13:00   13.400000   
1    2024-07-01 08:13:00 2024-07-01 08:13:00    0.183333   
2    2024-07-01 08:13:00 2024-07-01 08:18:00    4.433333   
3    2024-07-01 08:18:00 2024-07-01 09:17:00   59.100000   
4    2024-07-01 09:17:00 2024-07-01 09:18:00    1.016666   
...                  ...                 ...         ...   
6639 2024-07-31 00:18:00 2024-07-31 00:48:00   30.016666   
6640 2024-07-31 00:48:00 2024-07-31 00:50:00    2.000000   
6641 2024-07-31 00:50:00 2024-07-31 01:21:00   31.516666   
6642 2024-07-31 01:21:00 2024-07-31 01:22:00    0.316666   
6643 2024-07-31 01:22:00 2024-07-31 08:00:00  397.883333   

                       reason_x           downtime           reason_y  \
0            Производство > 95%                NaN                NaN   
1     Вмешательство оператора\n                NaN                NaN   
2      Производство > 75% < 95%                NaN                NaN   
3            Производство > 95%                NaN                NaN   
4                Внешняя ошибка                NaN                NaN   
...                         ...                ...                ...   
6639         Производство > 95%                NaN                NaN   
6640   Производство > 75% < 95%                NaN                NaN   
6641         Производство > 95%                NaN                NaN   
6642   Производство > 75% < 95%                NaN                NaN   
6643                    Останов  Недоступное время  Отсутствие заявок   

            description  reason_id detailed_description component  
0                   NaN        NaN                  NaN       NaN  
1                   NaN        NaN                  NaN       NaN  
2                   NaN        NaN                  NaN       NaN  
3                   NaN        NaN                  NaN       NaN  
4                   NaN        NaN                  NaN       NaN  
...                 ...        ...                  ...       ...  
6639                NaN        NaN                  NaN       NaN  
6640                NaN        NaN                  NaN       NaN  
6641                NaN        NaN                  NaN       NaN  
6642                NaN        NaN                  NaN       NaN  
6643  Отсутствие заявок  2913801.0   Отсутствие заявок.    Розлив  

[6644 rows x 10 columns]

In [40]:
delay_full_data.to_excel("Сконкатенированные простои.xlsx")

In [43]:
delay_full_data[delay_full_data["reason_x"] == "Вмешательство оператора\n"]

start                 end    duration  \
0   2024-07-01 11:35:00 2024-07-01 11:55:00   20.416666   
1   2024-07-01 15:52:00 2024-07-01 16:14:00   21.700000   
4   2024-07-01 18:51:00 2024-07-01 18:54:00    2.350000   
7   2024-07-01 22:33:00 2024-07-01 23:17:00   44.000000   
8   2024-07-01 23:57:00 2024-07-02 00:00:00    2.200000   
..                  ...                 ...         ...   
334 2024-07-29 23:55:00 2024-07-30 00:16:00   20.833333   
335 2024-07-30 03:37:00 2024-07-30 06:06:00  149.450000   
336 2024-07-30 10:24:00 2024-07-30 10:34:00    9.950000   
339 2024-07-30 13:49:00 2024-07-30 14:38:00   48.950000   
344 2024-07-31 00:09:00 2024-07-31 00:11:00    2.350000   

                      reason_x                    downtime  \
0    Вмешательство оператора\n                 Переналадки   
1    Вмешательство оператора\n                 Переналадки   
4    Вмешательство оператора\n  Прочие внеплановые простои   
7    Вмешательство оператора\n                 Переналадки   
8    Вмешательство оператора\n             Организационные   
..                         ...                         ...   
334  Вмешательство оператора\n             Организационные   
335  Вмешательство оператора\n                 Переналадки   
336  Вмешательство оператора\n                 Переналадки   
339  Вмешательство оператора\n       Санитарные регламенты   
344  Вмешательство оператора\n                 Переналадки   

                         reason_y                         description  \
0    Переход продукт вода продукт  Переход продукт вода продукт + АПМ   
1    Переход продукт вода продукт  Переход продукт вода продукт + АПМ   
4            Загруженность склада                             Съем ГП   
7    Переход продукт вода продукт  Переход продукт вода продукт + АПМ   
8                        Перерывы                Прогон в 00:00 часов   
..                            ...                                 ...   
334                      Перерывы                Прогон в 00:00 часов   
335          Переход через щ. СИП              Продукт СИП щ. продукт   
336                Смена этикетки       Смена дизайна (экспорт/промо)   
339                 Внешние мойки   АПМ (Пенная санитарная обработка)   
344          Переход через щ. СИП              Продукт СИП щ. продукт   

     reason_id         detailed_description component  
0      2214102                      переход    Розлив  
1      2214102                      переход    Розлив  
4      2112001                    нет сьема    Розлив  
7      2214102       переход на Флэш манго.    Розлив  
8      2511202               прогон партии.    Розлив  
..         ...                          ...       ...  
334    2511202            бакус дезинфекция    Розлив  
335    2214201                      переход    Розлив  
336    2214601                  Смена банки    Розлив  
339    2310503                          АПМ    Розлив  
344    2214201  Переход на FLash. СИП. АПМ.    Розлив  

[189 rows x 10 columns]

В данных за июль не оказалось полезной информации, переходим к датасету поломок.

In [5]:
df = pd.read_excel("Категоризированные поломки.xlsx")
df["reason_short"].value_counts()

reason_short
Замена тормозной пластины           140
Замена дросселя                      70
Замена двух дросселей                57
Устранение недоливов                 48
Замена сбросного дросселя            28
Замена нагнетательного дросселя      23
Иные поломки                         21
Замена наполнителя                   20
Замена дросселя CO2                  19
Замена верхнего дросселя             15
Замена прижимного колпака            14
Замена прижимного колокола           14
Замена нижнего дросселя              13
Замена дросселя нагрузки             13
Иные серьезные поломки               11
Замена прижимного стакана             9
Замена датчика                        9
КИП.МЕХ. Сбой пластичной смазки.      7
Name: count, dtype: int64

Сгенерируем признаки и таргет предсказания следующим образом:

* Сгенерируем временные метки с 2020 по 2024 года с разницей в 3 часа - объекты нашей выборки

* Для каждой временной метки посчитаем - сколько раз за последние X дней случалась ошибка Y, случалась ли она, и суммарное время простоев из-за ошибки. X берем из заранее заготовленного списка временных окон, которые нас интересуют - полгода, 3 месяца, месяц, 2 недели, неделя, 3 дня и один день.

* Тоже самое делаем для таргета, но теперь временные окна отложены не назад, а вперед. Также добавляем признак - какая ошибка будет следующей. 

На этих данных будем обучать нашу модель.

In [47]:
import pandas as pd

def generate_features_and_targets(df, time_range, window_sizes):
    # Создаем новые DataFrames для хранения признаков и таргетов
    feature_rows = []  # Список для хранения строк признаков
    target_rows = []   # Список для хранения строк таргетов
    
    for now in time_range:
        # Создаем строку для текущего времени
        new_feature_row = {'date': now}  # Заполняем временные рамки
        new_target_row = {'date': now}    # Заполняем временные рамки для таргетов
        
        for window in window_sizes:
            start_window = now - pd.Timedelta(days=window)
            
            for reason in df['reason_short'].unique():
                # Генерируем новые признаки
                if window > 0:
                    # Фильтруем строки по текущему reason_short и временным границам
                    mask = (df['reason_short'] == reason) & (df['start'] < now) & (df['end'] > start_window)
                    filtered_df = df[mask]
                    
                    new_feature_row[f'reason_{reason}_within_last_{window}_days'] = int(not filtered_df.empty)
                    new_feature_row[f'reason_{reason}_count_last_{window}_days'] = len(filtered_df)
                    new_feature_row[f'reason_{reason}_duration_sum_last_{window}_days'] = filtered_df['duration'].sum() if not filtered_df.empty else 0
                
                # Генерируем таргеты только для отрицательных окон
                if window < 0:
                    mask = (df['reason_short'] == reason) & (df['start'] > now) & (df['start'] < start_window)
                    filtered_df = df[mask]
                    
                    new_target_row[f'reason_{reason}_within_next_{-window}_days'] = int(not filtered_df.empty)
                    new_target_row[f'reason_{reason}_count_next_{-window}_days'] = len(filtered_df)
                    new_target_row[f'reason_{reason}_duration_sum_next_{-window}_days'] = filtered_df['duration'].sum() if not filtered_df.empty else 0
            
        new_target_row["next_breakdown"] = df[df["start"] > now].iloc[0]["reason_short"]
        
        # Добавляем новую строку в список признаков
        feature_rows.append(new_feature_row)
        
        target_rows.append(new_target_row)

    # Создаем DataFrame из списков строк
    feature_df = pd.DataFrame(feature_rows)
    target_df = pd.DataFrame(target_rows)

    return feature_df, target_df

# Устанавливаем начальную и конечную даты
start_date = '2020-07-01 08:00'
end_date = '2024-10-21 20:00'

# Генерация данных с частотой в 3 часа
time_range = pd.date_range(start=start_date, end=end_date, freq='3h')

feature_data, target_data = generate_features_and_targets(df, time_range, [180, 90, 30, 14, 7, 3, 1, -1, -3, -7, -14, -30])

C:\Users\user\AppData\Local\Temp\ipykernel_11964\3245234933.py:54: FutureWarning: 'H' is deprecated and will be removed in a future version, please use 'h' instead.
  time_range = pd.date_range(start=start_date, end=end_date, freq='3H')


In [48]:
feature_data.sample(10)

date  reason_Замена двух дросселей_within_last_180_days  \
3602  2021-09-24 14:00:00                                                  1   
11245 2024-05-06 23:00:00                                                  0   
6337  2022-09-01 11:00:00                                                  1   
2971  2021-07-07 17:00:00                                                  1   
4516  2022-01-16 20:00:00                                                  1   
8214  2023-04-24 02:00:00                                                  1   
7787  2023-03-01 17:00:00                                                  1   
5359  2022-05-02 05:00:00                                                  1   
6631  2022-10-08 05:00:00                                                  0   
7021  2022-11-25 23:00:00                                                  1   

       reason_Замена двух дросселей_count_last_180_days  \
3602                                                  8   
11245                                                 0   
6337                                                  4   
2971                                                  1   
4516                                                 12   
8214                                                 10   
7787                                                 11   
5359                                                 12   
6631                                                  0   
7021                                                  2   

       reason_Замена двух дросселей_duration_sum_last_180_days  \
3602                                           39.683333         
11245                                           0.000000         
6337                                           16.733333         
2971                                            8.633333         
4516                                           59.233333         
8214                                           36.416667         
7787                                           43.733333         
5359                                           52.283333         
6631                                            0.000000         
7021                                           10.700000         

       reason_Иные поломки_within_last_180_days  \
3602                                          0   
11245                                         0   
6337                                          1   
2971                                          1   
4516                                          0   
8214                                          0   
7787                                          1   
5359                                          1   
6631                                          1   
7021                                          1   

       reason_Иные поломки_count_last_180_days  \
3602                                         0   
11245                                        0   
6337                                         5   
2971                                         1   
4516                                         0   
8214                                         0   
7787                                         2   
5359                                         4   
6631                                         1   
7021                                         2   

       reason_Иные поломки_duration_sum_last_180_days  \
3602                                         0.000000   
11245                                        0.000000   
6337                                        33.950000   
2971                                         8.583333   
4516                                         0.000000   
8214                                         0.000000   
7787                                        11.333333   
5359                                        28.683333   
6631                                         5.266667   
7021                                        11.333333   

       reason_Замена н

In [49]:
target_data.sample(10)

date  reason_Замена двух дросселей_within_next_1_days  \
5237  2022-04-16 23:00:00                                                0   
11658 2024-06-27 14:00:00                                                0   
8630  2023-06-15 02:00:00                                                0   
5481  2022-05-17 11:00:00                                                0   
2000  2021-03-08 08:00:00                                                0   
3565  2021-09-19 23:00:00                                                0   
9106  2023-08-13 14:00:00                                                0   
9616  2023-10-16 08:00:00                                                0   
9413  2023-09-20 23:00:00                                                0   
7222  2022-12-21 02:00:00                                                0   

       reason_Замена двух дросселей_count_next_1_days  \
5237                                                0   
11658                                               0   
8630                                                0   
5481                                                0   
2000                                                0   
3565                                                0   
9106                                                0   
9616                                                0   
9413                                                0   
7222                                                0   

       reason_Замена двух дросселей_duration_sum_next_1_days  \
5237                                                 0.0       
11658                                                0.0       
8630                                                 0.0       
5481                                                 0.0       
2000                                                 0.0       
3565                                                 0.0       
9106                                                 0.0       
9616                                                 0.0       
9413                                                 0.0       
7222                                                 0.0       

       reason_Иные поломки_within_next_1_days  \
5237                                        0   
11658                                       0   
8630                                        0   
5481                                        0   
2000                                        0   
3565                                        0   
9106                                        0   
9616                                        0   
9413                                        0   
7222                                        0   

       reason_Иные поломки_count_next_1_days  \
5237                                       0   
11658                                      0   
8630                                       0   
5481                                       0   
2000                                       0   
3565                                       0   
9106                                       0   
9616                                       0   
9413                                       0   
7222                                       0   

       reason_Иные поломки_duration_sum_next_1_days  \
5237                                            0.0   
11658                                           0.0   
8630                                            0.0   
5481                                            0.0   
2000                                            0.0   
3565                                            0.0   
9106                                            0.0   
9616                                            0.0   
9413                                            0.0   
7222                                            0.0   

       reason_Замена нагнетательного дросселя_within_next_1_days  \
5237                                                   0           
11658             

In [50]:
feature_data.to_excel("Features_3H.xlsx")
target_data.to_excel("Target_3H.xlsx")

Для начала попробуем предсказывать следующую ошибку. Для начала нам нужно отобрать только данные по самым частым ошибкам, чтобы они встречались при обучении:

In [108]:
target_data["next_breakdown"].value_counts()

next_breakdown
Замена тормозной пластины           2928
Устранение недоливов                1648
Замена двух дросселей               1145
Замена дросселя                     1141
Замена сбросного дросселя           1009
Иные поломки                         811
Замена дросселя CO2                  682
Замена датчика                       630
Замена верхнего дросселя             622
Замена дросселя нагрузки             405
Замена нагнетательного дросселя      347
КИП.МЕХ. Сбой пластичной смазки.     341
Иные серьезные поломки               303
Замена прижимного стакана            210
Замена прижимного колокола           126
Замена наполнителя                   116
Замена нижнего дросселя              108
Замена прижимного колпака             17
Name: count, dtype: int64

Выберем четыре самых популярных значения

In [112]:
top_9_breakdowns = target_data["next_breakdown"].value_counts().nlargest(4).index
target_data_red = target_data[target_data["next_breakdown"].isin(top_9_breakdowns)]

feature_data_red = feature_data.loc[feature_data.index.isin(target_data_red.index)]

In [116]:
from sklearn.preprocessing import LabelEncoder

# Загрузка данных
X = feature_data_red.drop(columns=["date"])

# Преобразуем события в числовые метки
event_encoder = LabelEncoder()
y = event_encoder.fit_transform(target_data_red["next_breakdown"])

# Разделение данных на обучающую и тестовую выборки
train_size = int(0.8 * len(X))
X_train, X_test, y_train, y_test = X[:3 * train_size // 4], X[3 * train_size // 4:train_size], y[:3 * train_size // 4], y[3 * train_size // 4:train_size]

In [124]:
from sklearn.preprocessing import LabelBinarizer
from sklearn.metrics import roc_auc_score
import optuna
from optuna.samplers import TPESampler
import lightgbm as lgb

# Преобразование y в бинарный формат
lb = LabelBinarizer()
y_train_binarized = lb.fit_transform(y_train)
y_test_binarized = lb.transform(y_test)

# Функция для Optuna
def objective(trial):
    n_estimators = trial.suggest_int('n_estimators', 50, 200)
    max_depth = trial.suggest_int('max_depth', 10, 50)
    learning_rate = trial.suggest_float('learning_rate', 0.01, 0.1)

    model = lgb.LGBMClassifier(n_estimators=n_estimators, max_depth=max_depth, learning_rate=learning_rate, random_state=239, verbose=0)
    
    model.fit(X_train, y_train)
    y_pred_proba = model.predict_proba(X_test)[:]
    return roc_auc_score(y_test_binarized, y_pred_proba, multi_class='ovr')

study = optuna.create_study(direction='maximize', sampler=TPESampler())
study.optimize(objective, n_trials=50)

# Лучшие параметры
best_params = study.best_params
print(f'Best ROC-AUC: {study.best_value:.2f}')
print(f'Best parameters: {best_params}')

# Обучение модели с лучшими параметрами
model = lgb.LGBMClassifier(**best_params, random_state=42)

model.fit(X_train, y_train)


[I 2025-02-24 21:17:30,038] A new study created in memory with name: no-name-ddd227b6-ad44-4ba7-ba14-63efd6cb04ae


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:30,824] Trial 0 finished with value: 0.661092086318237 and parameters: {'n_estimators': 94, 'max_depth': 50, 'learning_rate': 0.03614309984660987}. Best is trial 0 with value: 0.661092086318237.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:31,614] Trial 1 finished with value: 0.6382517597439485 and parameters: {'n_estimators': 103, 'max_depth': 19, 'learning_rate': 0.07786194948216822}. Best is trial 0 with value: 0.661092086318237.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:32,671] Trial 2 finished with value: 0.6604250606143678 and parameters: {'n_estimators': 145, 'max_depth': 44, 'learning_rate': 0.06698909106824008}. Best is trial 0 with value: 0.661092086318237.


[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No f

[I 2025-02-24 21:17:33,719] Trial 3 finished with value: 0.6405937875116514 and parameters: {'n_estimators': 160, 'max_depth': 29, 'learning_rate': 0.08405760702604036}. Best is trial 0 with value: 0.661092086318237.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:34,534] Trial 4 finished with value: 0.6691966453048404 and parameters: {'n_estimators': 113, 'max_depth': 18, 'learning_rate': 0.04739126527612814}. Best is trial 4 with value: 0.6691966453048404.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:35,413] Trial 5 finished with value: 0.6611794336548258 and parameters: {'n_estimators': 121, 'max_depth': 37, 'learning_rate': 0.031360332682817106}. Best is trial 4 with value: 0.6691966453048404.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning]

[I 2025-02-24 21:17:36,429] Trial 6 finished with value: 0.6207115471628898 and parameters: {'n_estimators': 181, 'max_depth': 23, 'learning_rate': 0.08520431880424842}. Best is trial 4 with value: 0.6691966453048404.


[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No f

[I 2025-02-24 21:17:37,332] Trial 7 finished with value: 0.7172201153621615 and parameters: {'n_estimators': 128, 'max_depth': 14, 'learning_rate': 0.049043198809909934}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:37,796] Trial 8 finished with value: 0.6664437558507577 and parameters: {'n_estimators': 59, 'max_depth': 30, 'learning_rate': 0.06502279383135837}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:38,476] Trial 9 finished with value: 0.642857203944096 and parameters: {'n_estimators': 84, 'max_depth': 21, 'learning_rate': 0.07315928169221679}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:39,963] Trial 10 finished with value: 0.6765809778543721 and parameters: {'n_estimators': 200, 'max_depth': 10, 'learning_rate': 0.014944631124885782}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:41,377] Trial 11 finished with value: 0.6846546191068363 and parameters: {'n_estimators': 194, 'max_depth': 11, 'learning_rate': 0.02579912909524263}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:42,388] Trial 12 finished with value: 0.4887670528484661 and parameters: {'n_estimators': 149, 'max_depth': 10, 'learning_rate': 0.010232999203174255}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning]

[I 2025-02-24 21:17:43,367] Trial 13 finished with value: 0.5926870217347648 and parameters: {'n_estimators': 175, 'max_depth': 14, 'learning_rate': 0.0995314610534361}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No f

[I 2025-02-24 21:17:44,816] Trial 14 finished with value: 0.6193185659122034 and parameters: {'n_estimators': 200, 'max_depth': 25, 'learning_rate': 0.04780258124488478}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No f

[I 2025-02-24 21:17:45,792] Trial 15 finished with value: 0.6516809069230177 and parameters: {'n_estimators': 135, 'max_depth': 15, 'learning_rate': 0.026588906558883896}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:46,320] Trial 16 finished with value: 0.6893495540250789 and parameters: {'n_estimators': 66, 'max_depth': 36, 'learning_rate': 0.054975211518824785}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:46,810] Trial 17 finished with value: 0.6753729035355942 and parameters: {'n_estimators': 60, 'max_depth': 37, 'learning_rate': 0.056158333494536986}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:47,409] Trial 18 finished with value: 0.6631291138727622 and parameters: {'n_estimators': 77, 'max_depth': 34, 'learning_rate': 0.04297479758829036}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:48,372] Trial 19 finished with value: 0.648052466826716 and parameters: {'n_estimators': 129, 'max_depth': 44, 'learning_rate': 0.05792802813441576}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:48,791] Trial 20 finished with value: 0.5358347174067569 and parameters: {'n_estimators': 51, 'max_depth': 27, 'learning_rate': 0.03936851600144475}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:50,023] Trial 21 finished with value: 0.6525147117293444 and parameters: {'n_estimators': 172, 'max_depth': 14, 'learning_rate': 0.02147994644291679}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:50,637] Trial 22 finished with value: 0.6393802330288236 and parameters: {'n_estimators': 76, 'max_depth': 33, 'learning_rate': 0.05022475567339499}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:51,404] Trial 23 finished with value: 0.5527430928136656 and parameters: {'n_estimators': 104, 'max_depth': 16, 'learning_rate': 0.0240766821325397}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:52,584] Trial 24 finished with value: 0.6352805080021399 and parameters: {'n_estimators': 160, 'max_depth': 41, 'learning_rate': 0.060167631799573754}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No f

[I 2025-02-24 21:17:53,944] Trial 25 finished with value: 0.653653780339608 and parameters: {'n_estimators': 186, 'max_depth': 13, 'learning_rate': 0.03365299758411352}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:55,218] Trial 26 finished with value: 0.6708622069542399 and parameters: {'n_estimators': 142, 'max_depth': 22, 'learning_rate': 0.05097301482889781}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:56,231] Trial 27 finished with value: 0.7099506370148496 and parameters: {'n_estimators': 118, 'max_depth': 10, 'learning_rate': 0.042179406871894726}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:57,202] Trial 28 finished with value: 0.6690007600873026 and parameters: {'n_estimators': 119, 'max_depth': 19, 'learning_rate': 0.04334204736494774}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:57,998] Trial 29 finished with value: 0.6589448860217588 and parameters: {'n_estimators': 97, 'max_depth': 50, 'learning_rate': 0.035372360289655985}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:58,709] Trial 30 finished with value: 0.7155354056266781 and parameters: {'n_estimators': 88, 'max_depth': 38, 'learning_rate': 0.06843034301229423}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:17:59,417] Trial 31 finished with value: 0.7075791311846629 and parameters: {'n_estimators': 88, 'max_depth': 39, 'learning_rate': 0.06382330362387639}. Best is trial 7 with value: 0.7172201153621615.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:00,147] Trial 32 finished with value: 0.7216698197265672 and parameters: {'n_estimators': 89, 'max_depth': 40, 'learning_rate': 0.06891772081220097}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:01,020] Trial 33 finished with value: 0.6465908808523158 and parameters: {'n_estimators': 109, 'max_depth': 46, 'learning_rate': 0.06988073677746}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:01,796] Trial 34 finished with value: 0.5802452448208597 and parameters: {'n_estimators': 92, 'max_depth': 41, 'learning_rate': 0.08006461718767484}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:02,545] Trial 35 finished with value: 0.6582210638607809 and parameters: {'n_estimators': 98, 'max_depth': 33, 'learning_rate': 0.08741883254515916}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:03,394] Trial 36 finished with value: 0.6522083998704687 and parameters: {'n_estimators': 111, 'max_depth': 45, 'learning_rate': 0.07556092382942554}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:04,522] Trial 37 finished with value: 0.6721621504297712 and parameters: {'n_estimators': 131, 'max_depth': 42, 'learning_rate': 0.06254989165652693}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:05,429] Trial 38 finished with value: 0.6675267950634446 and parameters: {'n_estimators': 122, 'max_depth': 50, 'learning_rate': 0.06792096185490723}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:06,054] Trial 39 finished with value: 0.6788249346020014 and parameters: {'n_estimators': 80, 'max_depth': 17, 'learning_rate': 0.08034842232055076}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:06,872] Trial 40 finished with value: 0.6583242888600471 and parameters: {'n_estimators': 105, 'max_depth': 30, 'learning_rate': 0.0716540540773256}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:07,544] Trial 41 finished with value: 0.6449742323701761 and parameters: {'n_estimators': 88, 'max_depth': 39, 'learning_rate': 0.06582567301553924}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:08,096] Trial 42 finished with value: 0.6616907131411164 and parameters: {'n_estimators': 71, 'max_depth': 39, 'learning_rate': 0.053483224881136576}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:08,765] Trial 43 finished with value: 0.6650549102631412 and parameters: {'n_estimators': 88, 'max_depth': 48, 'learning_rate': 0.0624239848388728}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:09,652] Trial 44 finished with value: 0.6218225212542285 and parameters: {'n_estimators': 113, 'max_depth': 35, 'learning_rate': 0.09146943005935389}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No f

[I 2025-02-24 21:18:10,416] Trial 45 finished with value: 0.6559655674059001 and parameters: {'n_estimators': 97, 'max_depth': 39, 'learning_rate': 0.04219557625508366}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning]

[I 2025-02-24 21:18:11,525] Trial 46 finished with value: 0.6335587303347201 and parameters: {'n_estimators': 141, 'max_depth': 42, 'learning_rate': 0.07417769584107561}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:12,750] Trial 47 finished with value: 0.6426972212725235 and parameters: {'n_estimators': 153, 'max_depth': 12, 'learning_rate': 0.04689972821720864}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:13,698] Trial 48 finished with value: 0.6377613733546306 and parameters: {'n_estimators': 118, 'max_depth': 32, 'learning_rate': 0.06106947927075509}. Best is trial 32 with value: 0.7216698197265672.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


[I 2025-02-24 21:18:14,398] Trial 49 finished with value: 0.5575029286401235 and parameters: {'n_estimators': 84, 'max_depth': 37, 'learning_rate': 0.029471373463378435}. Best is trial 32 with value: 0.7216698197265672.


Best ROC-AUC: 0.72
Best parameters: {'n_estimators': 89, 'max_depth': 40, 'learning_rate': 0.06891772081220097}
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf


LGBMClassifier(learning_rate=0.06891772081220097, max_depth=40, n_estimators=89,
               random_state=42)

Итак, мы получили ROC-AUC score 0.72 на исторических данных, что хорошо для задач такого типа. Приналичии большего количества донных и более унифицированной специализации ошибок можно обучить эту модель заново и получить результаты еще лучше, а также модель сможет предсказывать более широкий спектр ошибок.